pallets/deitos/src/lib.rs

```diff
@@ -80,6 +80,10 @@ pub mod pallet {
         #[pallet::constant]
         type MaxPaymentPlanDuration: Get<u32>;
 
+        /// Maximum number of agreements per consumer
+        #[pallet::constant]
+        type MaxAgreements: Get<u32>;
```

Comment:
Yes it's used here:

`pub active_agreements: ActiveAgreements<T>,`

However if we want to add this to enhance queries performance i agree it would be better to define a separate storage item to like `ip -> agreement_id`